File: frontend/src/api/departmentAPI.js
import { http } from './http.js';

export async function getDepartmentLoad(){
    try {
        const data = await http ('/departments/load');
        return data;
    } catch (err) {
        return {
            success: false,
            message: err.message
        };
    }
}

export async function getDepartment(id){
    try {
        const data = await http (`/departments/${id}`, 
        {
            method: "Get",
        });
        return data;
    } catch (err) {
        return {
            success: false,
            message: err.message
        };
    }
}

export async function createDepartment(name){
    if(typeof name !== 'string' || name.trim() === ''){
        return {
            success: false,
            message: 'Department name must be a non-empty string'
        };
    }
    try{
        const data = await http (`/departments`, {
            method: "POST",
            body: JSON.stringify({name})
        });
        return data;
    } catch (err) {
        return {
            success: false,
            message: err.message
        };
    }
}

export async function updateDepartment(id, name){
    if(typeof name !== 'string' || name.trim() === ''){
        return {
            success: false,
            message: 'Department name must be a non-empty string'
        };
    }
     try{
        const data = await http (`/departments/${id}`, {
            method: "PUT",
            body: JSON.stringify({name})
        });
        return data;
    } catch (err) {
        return {
            success: false,
            message: err.message
        };
    }
}

export async function deleteDepartment(id) {
    try{
        const data = await http (`/departments/${id}`, {
            method: "DELETE",
        });
        return data;
    } catch (err) {
        return {
            success: false,
            message: err.message
        };
    }
}

File: docmanage_mobile/lib/api/department_api.dart
import 'http_service.dart';

class DepartmentApi {
  static Future<dynamic> getDepartmentLoad() async {
    try {
      final response = await HttpService.request(
        '/departments/load',
        method: 'GET',
      );
      return response;
    } catch (e) {
      return {
        "success": false,
        "message": e.toString(),
      };
    }
  }

  static Future<dynamic> createDepartment(String name) async {
    try {
      final response = await HttpService.request(
        '/departments',
        method: 'POST',
        body: {'name': name},
      );
      return response;
    } catch (e) {
      return {
        "success": false,
        "message": e.toString(),
      };
    }
  }

  static Future<dynamic> deleteDepartment(String id) async {
    try {
      final response = await HttpService.request(
        '/departments/$id',
        method: 'DELETE',
      );
      return response;
    } catch (e) {
      return {
        "success": false,
        "message": e.toString(),
      };
    }
  }
}